src/components/VideoPlayer/VideoPlayer.tsx

```diff
@@ -0,0 +1,42 @@
+import React from 'react';
+import styled from 'styled-components';
+
+/**
+ * Interface for the VideoPlayerProps.
+ * @property {string} src - The source of the video.
+ * @property {string} [title] - The title of the video used for accessibilty purposes.
+ */
+export interface VideoPlayerProps {
+  src: string;
+  title?: string;
+}
+
+/**
+ * Styled component for rendering an iframe that span the full width of its parent div.
+ */
+const VideoiFrame = styled.iframe`
+  min-width: 100%;
+  aspect-ratio: 16 / 9;
+`;
+
+/**
+ * Renders a video player component.
+ *
+ * @param {VideoPlayerProps} props - The props for the video player component.
+ * @param {string} props.src - The source URL of the video.
+ * @param {string} props.title - The title of the video player.
+ * @returns {JSX.Element} The rendered video player component.
+ */
+export const VideoPlayer = ({
+  src,
+  title = 'Quran Journey - YouTube Video',
+}: VideoPlayerProps): JSX.Element => {
+  return (
+    <VideoiFrame
```

Comment:
Hmm, since this is an Iframe, this might limit flexibility in the future—just something to note. It's not worth adapting it right now for future potential features.